js/data/symbol_bucket.js

```diff
@@ -40,6 +40,8 @@ function SymbolBucket(options) {
 
     this.adjustedIconMaxSize = this.layer.getLayoutValue('icon-size', 18, zoomHistory);
     this.adjustedIconSize = this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory);
+
+    this.index = 0;
```

Comment:
`SymbolInstance`s for the same feature still need different values. We could do something like `k * 10000 + j` but that seems a bit messier.

The value doesn't need to be an index, it just needs to be a unique value we can generate the same way in both -js and -native.

I thought generating it from the position, angle and style properties but that just seems way more complicated.